ganga/GangaDirac/Lib/Splitters/OfflineGangaDiracSplitter.py

```diff
@@ -257,7 +257,7 @@ def calculateSiteSEMapping(file_replicas, uniqueSE, CE_to_SE_mapping, SE_to_CE_m
         SE_dict[lfn] = sitez
 
     # Remove the banned sites (CE) from the mappings
-    for iSE in CE_to_SE_mapping.keys():
+    for iSE in list(CE_to_SE_mapping.keys()):
```

Comment:
Why is this necessary? `.keys()` should return an iterable generator.

Could also do the following:
```suggestion
    for iSE in CE_to_SE_mapping:
```

Reply:
But you need a copy of the list if you start to remove items.

Reply:
Ah sure, I didn't read the following code, just this line.